modules/collection/programs/helix.nix

```
@@ -0,0 +1,107 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.modules)
+    mkIf
+    ;
+  inherit (lib.attrsets)
+    optionalAttrs
+    mapAttrs'
+    nameValuePair
+    ;
+  inherit (lib.options) mkOption mkEnableOption mkPackageOption;
+  inherit (lib.types) attrsOf;
+
+  toml = pkgs.formats.toml { };
+
+  mkThemes =
+    themes:
+    mapAttrs' (
+      name: value:
+      nameValuePair ".config/helix/themes/${name}.toml" {
+        source = (toml.generate "helix-theme-${name}.toml" value);
+      }
+    ) themes;
+
+  cfg = config.rum.programs.helix;
+in
+{
+  options.rum.programs.helix = {
+    enable = mkEnableOption "Helix";
+
+    package = mkPackageOption pkgs "helix" { };
+
+    settings = mkOption {
+      type = toml.type;
+      default = { };
+      example = {
+        editor = {
+          line-number = "relative";
+          editor.statusline.left = [
+            "mode"
+            "spinner"
+          ];
+        };
+      };
+      description = ''
+        The editor configuration converted into TOML and written to
+        `${config.directory}/.config/helix/config.toml`.
+        Please reference https://docs.helix-editor.com/editor.html
+        and https://docs.helix-editor.com/remapping.html for config
+        options.
+      '';
+    };
+
+    languageSettings = mkOption {
```

```suggestion
    languages = mkOption {
```

Probably just have this be `languages` to fit with the spec of having the option name correspond to the file it writes to.

---

Done.